programs/cleos/main.cpp

```diff
@@ -2347,19 +2363,41 @@ int main( int argc, char** argv ) {
            std::cerr << localized("WARNING: ") << wasmPath << localized(" doesn't look like a binary WASM file. Is it something else, like WAST? Trying anyways...") << std::endl;
         code_bytes = bytes(wasm.begin(), wasm.end());
 
+        if (code_bytes.size() > 0 && code_bytes.size() == old_wasm.size()) {
+           duplicate = (memcmp(&(code_bytes[0]), &(old_wasm[0]), code_bytes.size()) == 0);
```

Comment:
I think I'd prefer to move us away from lexicographical comparison for equality.   what do you think about using `duplicate = std::equal(code_bytes.begin(), code_bytes.end(), old_wasm.begin())`